File: Source/Web/src/utils/api/goal/models/GoalDto.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { Experience } from './Experience';
import type { GoalCategories } from './GoalCategories';
import type { LearningType } from './LearningType';
import type { TimeAvailabilityPerDay } from './TimeAvailabilityPerDay';
import type { TimeAvailabilityPerWeek } from './TimeAvailabilityPerWeek';
import type { UserAdvancement } from './UserAdvancement';

export type GoalDto = {
    id?: string;
    name?: string | null;
    category?: GoalCategories;
    userAdvancement?: UserAdvancement;
    timeAvailabilityPerDay?: TimeAvailabilityPerDay;
    timeAvailabilityPerWeek?: TimeAvailabilityPerWeek;
    startDate?: string;
    endDate?: string;
    experience?: Experience;
    learningType?: LearningType;
    userInput?: string | null;
};

File: Source/GoalApi/Goal/Dtos/GoalDto.cs
using GoalApi.Enums;

namespace GoalApi.Goal.Dtos;

public class GoalDto
{
    public Guid Id { get; init; }
    public GoalCategories Category { get; init; }
    public UserAdvancement UserAdvancement { get; init; }
    public TimeAvailabilityPerDay TimeAvailabilityPerDay { get; init; }
    public TimeAvailabilityPerWeek TimeAvailabilityPerWeek { get; init; }
    public DateTime StartDate { get; init; }
    public DateTime EndDate { get; init; }
    public Experience Experience { get; init; }
    
    public LearningType LearningType { get; init; }
    public string UserInput { get; init; }
}